pkg/cgroups/manager/manager.go

```diff
@@ -192,16 +212,18 @@ func (c cgroupImpl) GetCgroupID(cgroupPath string) (uint64, error) {
 	return cgroups.GetCgroupID(cgroupPath)
 }
 
-func initManager(provider cgroupPathProvider, cg cgroup, channelSize int) *CgroupManager {
+func initManager(provider cgroupPathProvider, cg cgroup, channelSize int, events chan podEventStatus) *CgroupManager {
 	m := &CgroupManager{
 		podMetadataById:           make(map[string]*podMetadata),
 		containerMetadataByCgrpId: make(map[uint64]*containerMetadata),
 		podEvents:                 make(chan podEvent, channelSize),
 		shutdown:                  make(chan struct{}),
+		metadataCache:             map[uint64]PodMetadata{},
 	}
 	m.cgroupsChecker = cg
 	m.checkPathProvider = new(sync.Once)
 	m.pathProvider = provider
+	m.podEventsDone = events
```

Comment:
Should this channel be closed in the `<-m.shutdown` case in `processPodEvents()`?

Reply:
Yes, good spot.